### PR TITLE
Enable hit debugger push in Assurance

### PR DESCRIPTION
**Pull request by cacheung**
Enable hit debugger push in Assurance by adding X-Adobe-AEP-Validation-Token to the request header.  
Also removed p.debug=true in the call which is no longer needed.

Here is a screenshot of the request header for trackAction with Assurance is enabled in the app.  X-Adobe-AEP-Validation-Token is added in the header.  When Assurance is not enabled, it won't be sent with the header.
requestHeader

HitDebugger Hits we got after the feature

hitdebugger

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessorTests.kt [hunk 1]
```diff
@@ -333,8 +349,21 @@ class AnalyticsHitProcessorTests {
...
+        Mockito.`when`(
+            mockedExtensionApi.getSharedState(
+                Mockito.eq("com.adobe.assurance"),
+                Mockito.eq(null),
+                Mockito.eq(false),
+                Mockito.eq(SharedStateResolution.LAST_SET)
+            )
+        ).thenReturn(mockAssuranceSharedStateResult)
         Mockito.`when`(mockedAnalyticsState.isAssuranceSessionActive).thenReturn(true)
+        updateMockedSharedState("com.adobe.assurance", mapOf("integrationid" to "testIntegrationId"))
```
Comment (addb) [inline code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessorTests.kt:364]: I don't think this is needed. Remove the updateSharedstate method and mockedSharedState variable.

File: code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessorTests.kt [hunk 2]
```diff
@@ -333,8 +349,21 @@ class AnalyticsHitProcessorTests {
...
 
+        Thread.sleep(500L)
```
Comment (addb) [inline code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessorTests.kt:366]: Remove this sleep statement as well

**Comment by addb** [comment]
Can you paste the headers or Charles screenshot showing the headers?

File: code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessorTests.kt [hunk 3]
```diff
@@ -333,8 +349,21 @@ class AnalyticsHitProcessorTests {
 
     @Test
     fun `network success - assurance enabled`() {
+        // Set up mock Assurance Shared State
+        val mockAssuranceSharedStateData = mapOf("integrationid" to "testIntegrationId")
```
Comment (addb) [inline code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessorTests.kt:353]: Add more tests for negative cases:

- No integrationid in sharedstate
- Null integrationid in sharedstate
- Empty String integrationid in sharedstate

Comment (addb) [reply code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsHitProcessorTests.kt:353]: Also the case showing
will not be able to extract the integration id when sharedState status is `Pending`

**Review by addb — APPROVED** [review]

**Comment by cacheung** [comment]
> We are able to see hitDebugger responses right?

Yes, I attached the Assurance Event View with the hitDebugger response.
